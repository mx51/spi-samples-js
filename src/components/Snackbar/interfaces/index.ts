export interface ISnackbar {
  duration?: number;
  horizontal?: 'center' | 'left' | 'right';
  vertical?: 'bottom' | 'top';
}
