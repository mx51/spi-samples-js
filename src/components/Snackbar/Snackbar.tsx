import React from 'react';
import Snackbar from '@material-ui/core/Snackbar';
import { useAppDispatch, useAppSelector } from '../../redux/hooks';
import { selectedPairError } from '../../redux/reducers/PairFormSlice/PairFormSelectors';
import { readTerminalPairError } from '../../redux/reducers/PairFormSlice/pairFormSlice';
import { ISnackbar } from './interfaces';

export default function SnackbarWrapper({
  duration = 10000,
  horizontal = 'center',
  vertical = 'bottom',
}: ISnackbar): React.ReactElement {
  const dispatch = useAppDispatch();
  const error = useAppSelector(selectedPairError);

  const handleClose = (): void => {
    dispatch(
      readTerminalPairError({
        isShown: false,
        message: '',
      })
    );
  };

  return error?.isShown ? (
    <Snackbar
      anchorOrigin={{ vertical, horizontal }}
      autoHideDuration={duration}
      key={vertical + horizontal}
      message={error?.message}
      open={error?.isShown}
      onClose={handleClose}
    />
  ) : (
    <span />
  );
}
