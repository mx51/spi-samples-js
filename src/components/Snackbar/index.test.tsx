import React from 'react';
import { cleanup } from '@testing-library/react';
import Snackbar from '.';
import mockWithRedux from '../../utils/tests/common';

describe('Test <Snackbar />', () => {
  afterEach(cleanup);

  test('should match Snackbar snapshot test', () => {
    // Arrange
    const container = mockWithRedux(<Snackbar />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
