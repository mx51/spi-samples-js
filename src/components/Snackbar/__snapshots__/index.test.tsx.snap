// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Snackbar /> should match Snackbar snapshot test 1`] = `
<div>
  <span />
</div>
`;
